src/datasources/system/types.ts

```diff
@@ -1,6 +1,15 @@
 import { DataQuery } from '@grafana/schema'
 
+export enum QueryType {
+  Metadata = "Metadata",
+  Summary = "Summary"
+}
+
 export interface SystemQuery extends DataQuery {
-  queryText?: string;
-  constant: number;
+  queryClass: QueryType
+}
+
```

Comment:
It's good practice to have a default query populated as soon as the user opens the editor instead of having nothing selected:
![image](https://github.com/ni/systemlink-grafana-plugins/assets/9257800/79ffc2d2-8d87-460a-98c4-53f22a361c7a)

You can solve this by implementing the `getDefaultQuery` method. See the data frame data source for an example:

https://github.com/ni/systemlink-grafana-plugins/blob/2b21199edd6ce17caa86b65fadd213aafbda75f3/src/datasources/data-frame/DataFrameDataSource.ts#L77-L79

(I actually took a look at the grafana source, and the `_.clone` is overkill, so you can omit it). I think "Summary" makes sense as the default type.